### PR TITLE
feat: Add `not_found` to guild member chunk data

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
Implements `not_found` which was added in https://github.com/discord/discord-api-docs/pull/1078.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -229,6 +229,8 @@ declare module 'node:events' {
   }
 }
 
+type JSONValue = string | number | object | boolean | JSONValue[];
```
Comment (kyranet) [inline packages/discord.js/typings/index.d.ts:232]: Shouldn't this be exported?

Comment (Jiralite) [reply packages/discord.js/typings/index.d.ts:232]: Sure, I just assumed it was for internal use only.

Comment (kyranet) [reply packages/discord.js/typings/index.d.ts]: Well, it's documented in the docs without `@private`, but people may find it useful to have for handling the field's type when handling `notFound`.

Comment (Jiralite) [reply packages/discord.js/typings/index.d.ts]: Just exported the type and sorted it into the typedef region in the types!

**Review by kyranet — APPROVED** [review]

**Comment by jaw0r3k** [comment]
Isnt it an array of ids? 
image

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**